apps/extension/src/service-worker.ts

```diff
@@ -1,111 +1,97 @@
-import { localExtStorage } from '@penumbra-zone/storage';
+import { createProxyImpl } from '@penumbra-zone/transport/src/proxy';
+import { connectChromeRuntimeAdapter } from '@penumbra-zone/transport/src/chrome-runtime/adapter';
+
+import { ConnectRouter, createContextValues, createPromiseClient } from '@connectrpc/connect';
+import { createGrpcWebTransport } from '@connectrpc/connect-web';
+
+// this is a remote service we proxy
+import { Query as IbcClientService } from '@buf/cosmos_ibc.connectrpc_es/ibc/core/client/v1/query_connect';
+
+// these are local services we implement
+import { CustodyProtocolService } from '@buf/penumbra-zone_penumbra.connectrpc_es/penumbra/custody/v1alpha1/custody_connect';
+import { ViewProtocolService } from '@buf/penumbra-zone_penumbra.connectrpc_es/penumbra/view/v1alpha1/view_connect';
+import custodyImpl from '@penumbra-zone/router/src/grpc/custody';
+import viewImpl from '@penumbra-zone/router/src/grpc/view-protocol-server';
+
+// legacy stdRouter
+import { stdRouter } from '@penumbra-zone/router/src/std/router';
+import { isStdRequest } from '@penumbra-zone/types';
 
-const config = {
-  grpcEndpoint: await localExtStorage.get('grpcEndpoint'),
+// configure and initialize extension services. services are passed directly to stdRouter, and to rpc handlers as context
+const grpcEndpoint = await localExtStorage.get('grpcEndpoint');
+const servicesConfig = {
+  grpcEndpoint,
   getWallet: async () => {
     const wallets = await localExtStorage.get('wallets');
     if (!wallets.length) throw new Error('No wallets connected');
     const { fullViewingKey, id } = wallets[0]!;
     return { walletId: id, fullViewingKey };
   },
-} satisfies ServicesConfig;
-
-export const existingServices = new Services(config);
-await existingServices.initialize();
-
-/*
- * std router
- */
-const penumbraMessageHandler = (
-  message: unknown,
-  _sender: chrome.runtime.MessageSender,
-  sendResponse: (response: unknown) => void,
-) => {
-  if (!isStdRequest(message)) return;
-  stdRouter(message, sendResponse, existingServices);
-  return true;
 };
+const services = new Services(servicesConfig);
+await services.initialize();
 
-chrome.runtime.onMessage.addListener(penumbraMessageHandler);
-
-/*
- * adapter to existing router
- */
-const adaptOldRouter = (service: ServiceType) => {
-  return async (req: JsonValue): Promise<JsonValue | ReadableStream<JsonValue>> => {
-    const packed = Any.fromJson(req, { typeRegistry });
-    const unpacked = packed.unpack(typeRegistry);
-    if (!unpacked) throw Error('Recieved message of unknown type');
-    const sequence = performance.now();
+// this only handles stdClient requests
+chrome.runtime.onMessage.addListener(
+  (
+    message: unknown,
+    _: chrome.runtime.MessageSender,
+    sendResponse: (response: unknown) => void,
+  ) => {
+    if (!isStdRequest(message)) return;
+    stdRouter(message, sendResponse, services);
+    return true;
+  },
+);
```

Comment:
not much remaining that still uses stdClient. would like popup control to be independent, and 'sync' is a thing that probably doesn't need a specific request. those bits feel out of scope for this pr